resdk/tests/functional/data_upload/e2e_upload.py

```diff
@@ -0,0 +1,268 @@
+                    'OTHER_CHAR_2:subdural cortical tumor, frontal/lateral'
+                    ' location. Easily isolated sample',
+                    'TISSUE:Tumor'
+                ],
+                'strain': '',
+                'source': 'Tumor',
+                'organism': 'Rattus norvegicus',
+                'molecule': 'total RNA',
+                'annotator': 'Tristan Brown',
+                'description': '',
+            }
+        }
+
+        ann_2 = {}
+        # # Restore if duplicate samples may be annotated.
+        # ann_2 = {
+        #     'sample': {
+        #         'genotype': '',
+        #         'cell_type':
+        #         'Mixed',
+        #         'optional_char': [
+        #             'LIBRARY_STRATEGY:Illumina Standard Prep ',
+        #             'OTHER_CHAR_1:2x75 bp',
+        #             'OTHER_CHAR_2:subdural cortical tumor, frontal/lateral'
+        #             ' location. Easily isolated sample',
+        #             'TISSUE:Tumor'
+        #         ],
+        #         'strain': '',
+        #         'source': 'Tumor',
+        #         'organism': 'Homo sapiens',
+        #         'molecule': 'total RNA',
+        #         'annotator': 'Tristan Brown',
+        #         'description': '',
+        #     }
+        # }
+
+        ann_3 = {
+            'sample': {
+                'genotype': 'AX4',
+                'cell_type': '',
+                'optional_char': [
+                    'LIBRARY_STRATEGY:Illumina Standard Prep ',
+                    'OTHER_CHAR_1:300 bp'
+                ],
+                'strain': 'Non-aggregating',
+                'source': 'Cell',
+                'organism': 'Dictyostelium discoideum',
+                'molecule': 'genomic DNA',
+                'annotator': 'Tristan Brown',
+                'description': '',
+            }
+        }
+
+        reads_ann_1 = {
+            'experiment_type': 'RNA-Seq',
+            'protocols': {'antibody_information': {'manufacturer': ''},
+                          'extract_protocol': 'Standard',
+                          'fragmentation_method': '',
+                          'growth_protocol': 'Standard media',
+                          'library_prep': 'Illumina',
+                          'treatment_protocol': 'Control'},
+            'reads_info': {'barcode': '', 'facility': '', 'instrument_type': ''}
+        }
+
+        # Check annotations
+        sample_1.update()
+        sample_2.update()
+        sample_2b.update()
+        sample_3.update()
+
+        self.assertEqual(sample_1.descriptor, ann_1)
+        self.assertEqual(sample_1.data[0].descriptor, reads_ann_1)
+        self.assertEqual(sample_1.tags, ['community:rna-seq'])
+        self.assertEqual(sample_2.descriptor, ann_2)
+        self.assertEqual(sample_2b.descriptor, ann_2)
+        self.assertEqual(sample_3.descriptor, ann_3)
+        self.assertEqual(sample_4.descriptor, {})
```

Comment:
We should assert the logger message that the annotation is invalid as well.

Reply:
I don't know how to to do this, and I can't find prior examples of it in ReSDK.